web/templates/dialog_widget.hbs

```diff
@@ -12,6 +12,7 @@
             </header>
             <main class="modal__content" data-simplebar data-simplebar-tab-index="-1" {{#if always_visible_scrollbar}}data-simplebar-auto-hide="false"{{/if}}>
                 <div class="alert" id="dialog_error"></div>
+                <div id="image_loading_indicator"></div>
```

Comment:
nitpick: since you're using `dialog_widget` module, you can use the following points from `dialog_widget.ts`:
```
 *      5) If a modal wants a loading spinner, it should pass loading_spinner: true.
 *         This will show a loading spinner when the yes button is clicked.
 *         The caller is responsible for calling hide_confirm_dialog_spinner()
 *         to hide the spinner in both success and error handlers.
 *
 *      6) If loading_spinner is used, don't hide it on `success`. This modal has a fade out
 *         animation. This causes the `Confirm` button to be shown for a split second if the
 *         spinner is hidden.
 *         Just close the modal. This will remove the whole modal from the DOM without
 *         needing to remove the spinner.
 ```

Reply:
you can have a look at [other properties](https://github.com/zulip/zulip/blob/731391c01946bbfef987fc497ed9c34eb76065f1/web/src/dialog_widget.ts#L155) you can pass to `dialog_widget.launch`, if useful.

Reply:
Thank you for the information! From what I've read and tested, passing loading_spinner: true it enables a spinner inside of the submit button when clicked and that's not the desired behaviour. What we want is a loading spinner in the entire modal. Nonetheless, is there a better way to do this than the current implementation?